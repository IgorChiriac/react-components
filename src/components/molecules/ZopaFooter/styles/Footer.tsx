import styled from 'styled-components';
import { spacing } from '../../../../constants/spacing';
import grid from '../../../../constants/grid';
import { colors } from '../../../..';

export const Footer = styled.footer`
  backgroung-color: ${colors.white};
  padding-bottom: ${spacing[9]};
  padding-top: ${spacing[10]};

  @media (min-width: ${grid.breakpoints.l}px) {
    padding-top: ${spacing[11]};
  }
`;
