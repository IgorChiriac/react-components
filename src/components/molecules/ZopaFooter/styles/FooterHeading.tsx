import styled from 'styled-components';
import Heading from '../../../atoms/Heading/Heading';

export const FooterHeading = styled(Heading).attrs({ as: 'h3', size: 'h6' })`
  text-transform: uppercase;
`;
