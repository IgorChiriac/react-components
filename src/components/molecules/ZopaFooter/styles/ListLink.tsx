import React from 'react';
import styled from 'styled-components';
import Link, { LinkProps } from '../../../atoms/Link/Link';
import { typography } from '../../../../constants/typography';
import { spacing } from '../../../../constants/spacing';

const ListItem = styled.li`
  margin-bottom: ${spacing[4]};

  &:last-child {
    margin-bottom: 0;
  }
`;

const ListLinkWrapper = styled(Link)`
  font-weight: ${typography.weights.regular};
  text-decoration: none;

  &:hover,
  &:active {
    text-decoration: underline;
  }
`;

export const ListLink = (props: LinkProps) => (
  <ListItem>
    <ListLinkWrapper {...props} />
  </ListItem>
);
