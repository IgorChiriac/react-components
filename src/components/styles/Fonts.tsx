import { createGlobalStyle } from 'styled-components';

const Fonts = createGlobalStyle`
  /* Alverata */
  @font-face {
    font-family: 'Alverata';
    font-display: swap;
    /* IE9 Compat Modes */
    src: local('Alverata'), 
         url('https://s3-eu-west-1.amazonaws.com/cdn.zopa.com/fonts/1333416/2dc6ef4b-6ad7-4da6-b898-2ea5e967e712.eot');
    /* IE6-IE8 */
    src: local('Alverata'),
         url('https://s3-eu-west-1.amazonaws.com/cdn.zopa.com/fonts/1333416/2dc6ef4b-6ad7-4da6-b898-2ea5e967e712.eot?#iefix')
         format('eot'),
         /* Super Modern Browsers */
         url('https://s3-eu-west-1.amazonaws.com/cdn.zopa.com/fonts/1333416/42b8e48a-c882-4831-856e-4e1f148b3169.woff2')
         format('woff2'),
         /* Pretty Modern Browsers */
         url('https://s3-eu-west-1.amazonaws.com/cdn.zopa.com/fonts/1333416/99d874d5-3f1c-419c-9c82-4c63e62f98ca.woff')
         format('woff'),
         /* Safari, Android, iOS */
         url('https://s3-eu-west-1.amazonaws.com/cdn.zopa.com/fonts/1333416/c41ae233-742a-4c14-af50-6b0e62e12a58.ttf')
         format('truetype');
  }

  /* Open Sans */
  @import url('https://fonts.googleapis.com/css?family=Open+Sans:300,400,600,700');
`;

export default Fonts;
