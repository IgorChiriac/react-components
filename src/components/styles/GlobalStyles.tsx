import { createGlobalStyle, css } from 'styled-components';
import { normalize } from 'styled-normalize';
import { alverata, openSans } from '../../constants/fonts';

// Universal box sizing with Inheritance. More info: https://css-tricks.com/box-sizing/#article-header-id-6
const boxSizing = css`
  html {
    box-sizing: border-box;
  }
  *,
  *:before,
  *:after {
    box-sizing: inherit;
  }
`;

const defaultFont = css`
  html,
  body {
    font-family: ${openSans};
    font-size: 14px;
  }
  h1,
  h2,
  h3,
  h4,
  h5 {
    font-family: ${alverata};
  }
`;

const GlobalStyles = createGlobalStyle`
  ${normalize}
  ${defaultFont}
  ${boxSizing}
`;

export default GlobalStyles;
