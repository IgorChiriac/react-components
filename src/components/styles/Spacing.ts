import { css } from 'styled-components';
import { spacing as sizes } from '../../constants/spacing';
import grid, { TGridBreakpoints } from '../../constants/grid';

type TSpacingPositionNames = 'top' | 'right' | 'bottom' | 'left' | 'x' | 'y';
type TSpacingTypes = 'margin' | 'padding';
type TSpacingFactoryCallback = (pos: string, index: number, position: string, size: number) => string;

const positions: TSpacingPositionNames[] = ['top', 'right', 'bottom', 'left', 'x', 'y'];

const spacingCssFactory = (callback: TSpacingFactoryCallback): string =>
  sizes
    .map((sizingValue, sizingIndex) =>
      positions.map(position => callback(position.substr(0, 1), sizingIndex, position, sizingValue)).join(''),
    )
    .join(' ');

const propertyFactory = (type: TSpacingTypes, sizingValue: number, position: string): string => {
  if (position == 'y') {
    return `
        ${type}-top: ${sizingValue}px;
        ${type}-bottom: ${sizingValue}px;
      `;
  }

  if (position == 'x') {
    return `
        ${type}-left: ${sizingValue}px;
        ${type}-right: ${sizingValue}px;
      `;
  }

  return `${type}-${position}: ${sizingValue}px;`;
};

export const spacing = (type: TSpacingTypes) => css`

  ${sizes
    .map((sizingValue, sizingIndex) => `.${type.substr(0, 1)}-${sizingIndex} { ${type}: ${sizingValue}px; }`)
    .join('')}

  ${spacingCssFactory(
    (shortHandPosition, sizingIndex, position, sizingValue) => `
      .${type.substr(0, 1)}${shortHandPosition}-${sizingIndex} {
        ${propertyFactory(type, sizingValue, position)}
      }`,
  )}

  ${Object.keys(grid.breakpoints)
    .map(
      breakpoint => `
      @media screen and (min-width: ${grid.breakpoints[breakpoint as TGridBreakpoints]}px) {

        ${sizes
          .map(
            (sizingValue, sizingIndex) =>
              `.${breakpoint as TGridBreakpoints}${CSS.escape(':')}${type.substr(
                0,
                1,
              )}-${sizingIndex} { ${type}: ${sizingValue}px; }`,
          )
          .join('')}

        ${spacingCssFactory(
          (shortHandPosition, sizingIndex, position, sizingValue) => `
        .${breakpoint as TGridBreakpoints}${CSS.escape(':')}${type.substr(0, 1)}${shortHandPosition}-${sizingIndex} {
          ${propertyFactory(type, sizingValue, position)}
        }
      `,
        )}
      }
    `,
    )
    .join(' ')}
`;
