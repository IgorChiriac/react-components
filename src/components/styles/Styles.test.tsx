import React from 'react';
import { render } from '@testing-library/react';
import FontStyles from './Fonts';
import GlobalStyles from './GlobalStyles';
import { getGlobalStyleTags } from '../../helpers/test/styles';

it('allows to set global generic styles ', () => {
  render(<GlobalStyles />);

  const [globalGenericStyles] = getGlobalStyleTags();
  expect(globalGenericStyles).toMatchSnapshot();
});

it('allows to set global font styles', () => {
  render(<FontStyles />);

  const [globalFontStyles] = getGlobalStyleTags();
  expect(globalFontStyles).toMatchSnapshot();
});
