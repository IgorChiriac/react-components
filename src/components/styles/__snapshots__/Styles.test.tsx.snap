// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`allows to set global font styles 1`] = `"/* sc-component-id: sc-global-3277831321-import */ /* sc-component-id: sc-global-2771653525-import */ @import url('https://fonts.googleapis.com/css?family=Open+Sans:400,600,700&display=swap');"`;

exports[`allows to set global generic styles  1`] = `"/* sc-component-id: sc-global-3277831321-import */ @import url('https://unpkg.com/tippy.js@5.1.4/dist/tippy.css'); @import url('https://unpkg.com/tippy.js@5.1.4/animations/shift-toward.css');"`;
